libpromises/evalfunction.c

```diff
@@ -8601,6 +8666,16 @@ static const FnCallArg READFILE_ARGS[] =
     {NULL, CF_DATA_TYPE_NONE, NULL}
 };
 
+static const FnCallArg CLASSEXPRESSION_FILTERDATA_ARGS[] =
+{
+    {CF_ABSPATHRANGE, CF_DATA_TYPE_STRING, "File name to read"},
+    {CF_VALRANGE, CF_DATA_TYPE_INT, "Column with class expression"},
```

Comment:
Is this the column name or column index position?
If it is an index position, is it counting from 0 or 1?
If it is can be a column name, will it error if the column name is not found?

Reply:
Currently only column position. For the sake of consistency, it should definitely count from 0 like everything else (which is also what it does).

I am unsure if allowing both column position and name is a good idea. If column names are numbers, how will the function know whether to count the given parameter as a column index or name? If there are several matches of the column name, does it just go for the first match?

If this should be allowed, it should be decided by `has_heading`, but I would say it's better to only use indexes. Might be different for the `sort_by` column.

Reply:
+1 for only indices, keep it simple. I think you should update the description string to say:

```
"Column index with class expression"
```